frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/common/GHPRLoader.java

```diff
@@ -0,0 +1,40 @@
+package com.virtuslab.gitmachete.frontend.actions.common;
+
+import java.util.List;
+
+import com.intellij.openapi.project.Project;
+import lombok.val;
+import org.jetbrains.plugins.github.api.GHRepositoryCoordinates;
+import org.jetbrains.plugins.github.api.GithubServerPath;
+import org.jetbrains.plugins.github.api.data.pullrequest.GHPullRequestShort;
+import org.jetbrains.plugins.github.pullrequest.data.GHPRDataContextRepository;
+import org.jetbrains.plugins.github.util.GithubUrlUtil;
+
+import com.virtuslab.gitmachete.frontend.ui.providerservice.SelectedGitRepositoryProvider;
+
+public class GHPRLoader {
+
+  public static List<GHPullRequestShort> loadPRs(Project project) {
+    try {
+      val repository = GHPRDataContextRepository.Companion.getInstance(project);
+      val gitRepository = project.getService(SelectedGitRepositoryProvider.class)
+          .getSelectedGitRepository();
+      assert gitRepository != null;
+      val remotes = gitRepository.getRemotes();
+      val remote = remotes.iterator().next();
+      val url = remote.getFirstUrl();
+      assert url != null;
+      val repositoryPath = GithubUrlUtil.getUserAndRepositoryFromRemoteUrl(url);
```

Comment:
> 2. You can't control the PR search query, it's set by outside force

Amirite that you specifically mean that we're forced to use the fixed set of query methods like this `getUserAndRepositoryFromRemoteUrl`... rather than have a freedom to compose our own query (as it would be with HTTP GitHub API)? 🤔

Reply:
I meant literal outside force

When you open a pull request tab from GH plugin, there is a search box you can type in. This search box is the outside force.

I tried fixing it, but It seems that `internal` classes are not stable across versions. I have asked a relevant question on Slack.

Reply:
Huh okay... I'd advice then to look into the alternative way of dealing with GitHub PRs... probably some official Java SDK, if such thing exists? without the dependency on GitHub IntelliJ plugin :/